src/main/java/io/vertx/core/net/impl/ConnectionBase.java

```diff
@@ -468,18 +471,19 @@ public ChannelPromise channelFuture() {
   }
 
   public String remoteName() {
-    java.net.SocketAddress addr = chctx.channel().remoteAddress();
-    if (addr instanceof InetSocketAddress) {
-      // Use hostString that does not trigger a DNS resolution
-      return ((InetSocketAddress)addr).getHostString();
```

Comment:
In case of ha proxy we need to use the overriden address and not chctx.channel().remoteAddress

I was going to make another PR later to remove this method or at least deprecate it since i think it does exactly the same work as remoteAddress().host()

Please corrent me if i am wrong

Reply:
I mean the precise usage of `getHostString()` that has a purpose here to be not blocking

Reply:
remoteAddress().host() is exactly the same think.

```
public SocketAddressImpl(InetSocketAddress address) {
    Arguments.requireInRange(address.getPort(), 0, 65535, "port p must be in range 0 <= p <= 65535");
    this.path = null;
    this.port= address.getPort();
    this.host = address.getHostString();
    if (address.isUnresolved()) {
      this.hostName = address.getHostName();
      this.ipAddress = null;
    } else {
      String host = address.getHostString();
      if (NetUtil.isValidIpV4Address(host) || NetUtil.isValidIpV6Address(host)) {
        host = null;
      }
      this.hostName = host;
      this.ipAddress = address.getAddress();
    }
  }
```

Reply:
Now i get it. I assume that is the part that will block when using normal remoteAddress().host(). Right?
```
if (address.isUnresolved()) {
      this.hostName = address.getHostName();
      this.ipAddress = null;
} 
```